gtdynamics/universal_robot/Robot.h

```diff
@@ -112,6 +112,32 @@ class Robot {
   /// Print links and joints of the robot, for debug purposes
   void print(const std::string &s = "") const;
 
+  /// Overload equality operator.
+  bool operator==(const Robot &other) const {
+    // Define comparators for easy std::map equality checking
+    auto link_comparator = [](decltype(*this->name_to_link_.begin()) a,
+                              decltype(a) b) {
+      // compare the key name and the underlying shared_ptr object
+      return a.first == b.first && (*a.second) == (*b.second);
+    };
+    auto joint_comparator = [](decltype(*this->name_to_joint_.begin()) a,
+                               decltype(a) b) {
+      // compare the key name and the underlying shared_ptr object
+      return a.first == b.first && (*a.second) == (*b.second);
+    };
+
+    return (this->name_to_link_.size() == other.name_to_link_.size() &&
+            std::equal(this->name_to_link_.begin(), this->name_to_link_.end(),
+                       other.name_to_link_.begin(), link_comparator) &&
+            this->name_to_joint_.size() == other.name_to_joint_.size() &&
+            std::equal(this->name_to_joint_.begin(), this->name_to_joint_.end(),
+                       other.name_to_joint_.begin(), joint_comparator));
```

Comment:
note 1: `std::equal` is since c++14
note 2: I think if you give both the begin and end of both ranges and it'll automatically check the size so you don't have to?  apologies if you already tried this and have a reason it won't work

Reply:
1. `std::equal` is pre-C++11 :slightly_smiling_face: 
2. This version of `std::equal` which takes iterators to the start and end of both ranges is C++14.

Reply:
oh oops my mistake yep thanks :)